src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.Key.cs

```diff
@@ -50,6 +40,39 @@ private set
                 KeySize = key.KeySize;
             }
         }
+
+        public CngKey GenerateNewKey()
```

Comment:
Since this wasn't in the contract, is it just for testing?

I was expecting something like
```C#
public void GenerateNewKey(ECCurve curve)
{
   ...
}
```

So that "new" was a conscious decision.  But, we've done work to keep key generation as lazy as possible, so it should really just change the internal state to such that the key is invalid, and set the "next key curve" value.